oxidation/libparsec/crates/serialization_format/src/protocol.rs

```diff
@@ -556,11 +560,21 @@ fn quote_custom_enum(
                     }
                 }
                 Some(variant_fields) => {
-                    let variant_fields = quote_cmd_fields(variant_fields, false);
-                    quote! {
-                        #[serde(rename = #discriminant_value)]
-                        #variant_name {
-                            #(#variant_fields),*
+                    // The goal of untagged is to avoid fields like `{ foo: FooType }`
+                    // We want the inner value to be transparent
+                    if is_untagged {
+                        let variants_types = quote_variants_types(variant_fields);
+                        quote! {
+                            #[serde(rename = #discriminant_value)]
+                            #variant_name (#(#variants_types),*)
+                        }
```

Comment:
This actually generate a tuple variant

Reply:
I don't know if we should remove this code. At the very beginning I made `ActiveUsersLimit` an embedded type in schemas (we changed this here)[https://github.com/Scille/parsec-cloud/pull/3792#discussion_r1059427602].
It didn't break compatibility because an enum like this were generated.

```rs
#[derive(Serialize, Deserialize)]
#[serde(untagged)]
enum MyEnum {
    SomeValue(i32),
    Other,
}
// MyEnum::SomeValue(42) Serialize to "42" and MyEnum::Other gives `null`
```

Should we keep this or just remove it?

Reply:
I would state for keeping the minimal amount of change In a PR